paasta_tools/marathon_tools.py

```diff
@@ -911,6 +916,14 @@ def _namespaced_get_classic_service_information_for_nerve(name, namespace, soa_d
     port_file = os.path.join(soa_dir, name, 'port')
     nerve_dict['port'] = service_configuration_lib.read_port(port_file)
     nerve_name = compose_job_id(name, namespace)
+    try:
+        with open(_get_classic_service_puppet_file(name)) as extras:
+            if os.fstat(extras.fileno()).st_size > 0:
+                nerve_dict.update(json.load(extras))
+    except IOError:
+        log.exception("Could not open {0} for reading".format(extras.name))
+    except ValueError:
+        log.exception("Could not interpret {0} as JSON".format(extras.name))
     return (nerve_name, nerve_dict)
 
 
```

Comment:
Good question! I can use labels here to do what I want (explicitly making a label for tunnel_id & sharing them, instead of using the hostname). We only announce in one namespace at the moment, and I think our end-game is that the tunnel_id should be `service.namespace`.